app/assets/src/components/views/PipelineViz.jsx

```diff
@@ -276,11 +319,62 @@ class PipelineViz extends React.Component {
     });
   }
 
-  handleNodeHover(stageIndex, info) {
-    const { highlightColor } = this.props;
+  shouldUpdateMouseMove(x, y) {
```

Comment:
nit: this would sound more readable as shouldUpdateMoveMouse?

Reply:
if the goal is mostly to reduce the number of calls to handleMouseMove, you could also just use Lodash `throttle` https://css-tricks.com/debouncing-throttling-explained-examples/

It would save on this computation with every event firing

Reply:
I tried this at 100ms but it created a bit of a lag, especially when the mouse moves quickly — will look into it more in a separate PR since midpoint presentations are coming up soon 
![throttle](https://user-images.githubusercontent.com/23442055/60920899-1ed5fd00-a24e-11e9-841a-feb02c20db88.gif)